scipy/optimize/_nonlin.py

```diff
@@ -1491,9 +1493,39 @@ def __init__(self, rdiff=None, method='lgmres', inner_maxiter=20,
             self.method_kw.setdefault('store_outer_Av', False)
             self.method_kw.setdefault('atol', 0)
 
+        # Retrieve the signature of the method to find the valid parameters
+        valid_inner_params = [
+            k for k in signature(self.method).parameters
+            if k not in ('self', 'args', 'kwargs')
+        ]
+
         for key, value in kw.items():
-            if not key.startswith('inner_'):
+            if not key.startswith("inner_"):
                 raise ValueError(f"Unknown parameter {key}")
```

Comment:
I know this is pre existing but do you mind adding test coverage for this error while we are here